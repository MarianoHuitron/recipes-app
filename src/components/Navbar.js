import React, { useState } from 'react'
import { Link, useHistory } from 'react-router-dom';

const Navbar = () => {

    const [keySearch, setKeySearch] = useState("");
    const history = useHistory();

    const handleSubmit = e => {
        e.preventDefault();
        if(keySearch === "") return;
        history.push(`/search/${keySearch}`);

    }

    return (
        <nav className="navbar navbar-expand-leg navbar-dark bg-primary">
            <div className="container">
                <Link to="/" className="navbar-brand" >Recipes App</Link>
                    <form 
                        className="form-inline my-2 my-lg-0"
                        onSubmit={ handleSubmit }    
                    >
                        <input 
                            type="text" 
                            className="form-control mr-sm-2"
                            placeholder="Search by name"
                            value={ keySearch }
                            onChange={ (e) => setKeySearch(e.target.value) }    
                        />
                        <button
                            className="btn btn-secondary  my-sm-0"
                            type="submit"
                        >Search</button>
                    </form>
            </div>
        </nav>
    )
}

export default Navbar
