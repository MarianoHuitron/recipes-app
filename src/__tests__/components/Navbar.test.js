import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '../../components/Navbar';

const mockHistory = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: () => mockHistory()
    })
}));

describe('Testing <Navbar />', () => {
    
    const wrapper = shallow(<Navbar /> );

    test('should render component', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('simulate submit when input is empty', () => {
        wrapper.find('input').simulate('change', {
            target: {
                value: ''
            }
        });

        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        })

        expect( mockHistory ).not.toHaveBeenCalled();

    });

    test('simulate submit when input has change', () => {
        wrapper.find('input').simulate('change', {
            target: {
                value: 'Fish'
            }
        });

        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        })

        expect( mockHistory ).toHaveBeenCalled();

    })
    
    

})
