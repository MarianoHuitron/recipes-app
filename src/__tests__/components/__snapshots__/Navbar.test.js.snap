// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Navbar /> should render component 1`] = `
<nav
  className="navbar navbar-expand-leg navbar-dark bg-primary"
>
  <div
    className="container"
  >
    <Link
      className="navbar-brand"
      to="/"
    >
      Recipes App
    </Link>
    <form
      className="form-inline my-2 my-lg-0"
      onSubmit={[Function]}
    >
      <input
        className="form-control mr-sm-2"
        onChange={[Function]}
        placeholder="Search by name"
        type="text"
        value=""
      />
      <button
        className="btn btn-secondary  my-sm-0"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</nav>
`;
